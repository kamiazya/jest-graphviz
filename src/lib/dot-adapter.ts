import { execSync } from 'child_process';

function execCommand(command: string, input: string): Buffer {
  return execSync(command, {
    stdio: 'pipe',
    input,
  });
}

export function dotPlainStdin(input: string): string {
  return execCommand('dot -Tplain', input).toString();
}

export function dotJsonStdin(input: string): object {
  return JSON.parse(execCommand('dot -Tdot_json', input).toString());
}
