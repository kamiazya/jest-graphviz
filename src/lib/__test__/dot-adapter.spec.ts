import { dotJsonStdin, dotPlainStdin } from '../dot-adapter';

describe('dotPlainStdin test', () => {
  test('valid dot', () => {
    const dot = 'digraph g { a -> b; }';
    const result = dotPlainStdin(dot);
    expect(result).toMatchSnapshot();
  });

  test('invalid dot, to throw error', () => {
    const dot = 'invalid';
    expect(() => {
      dotPlainStdin(dot);
    }).toThrow();
  });
});

describe('dotJsonStdin test', () => {
  test('valid dot', () => {
    const dot = 'digraph g { a -> b; }';
    const result = dotJsonStdin(dot);
    expect(result).toMatchSnapshot();
  });

  test('invalid dot, to throw error', () => {
    const dot = 'invalid';
    expect(() => {
      dotJsonStdin(dot);
    }).toThrow();
  });
});
