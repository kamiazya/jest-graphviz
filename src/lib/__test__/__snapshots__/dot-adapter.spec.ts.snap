// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotJsonStdin test valid dot 1`] = `
Object {
  "_subgraph_cnt": 0,
  "directed": true,
  "edges": Array [
    Object {
      "_gvid": 0,
      "head": 1,
      "tail": 0,
    },
  ],
  "name": "g",
  "objects": Array [
    Object {
      "_gvid": 0,
      "label": "\\\\N",
      "name": "a",
    },
    Object {
      "_gvid": 1,
      "label": "\\\\N",
      "name": "b",
    },
  ],
  "strict": false,
}
`;

exports[`dotPlainStdin test valid dot 1`] = `
"graph 1 0.75 1.5
node a 0.375 1.25 0.75 0.5 a solid ellipse black lightgrey
node b 0.375 0.25 0.75 0.5 b solid ellipse black lightgrey
edge a b 4 0.375 0.99766 0.375 0.89071 0.375 0.76353 0.375 0.64468 solid black
stop
"
`;
